internal/ansible/dispatcher/dispatcher_events.go

```diff
@@ -0,0 +1,411 @@
+		Counter: msgCounter,
+		EventData: &message.AnsibleRunnerJobEventYamlEventData{
+			PlaybookUuid: &playbookUUID,
+			Changed:      message.AnsibleRunnerJobEventYamlEventDataChanged{},
+			Ok:           message.AnsibleRunnerJobEventYamlEventDataOk{},
+			Failures:     message.AnsibleRunnerJobEventYamlEventDataFailures{},
+		},
+	}
+	for host, stats := range stats {
+		changesMap := message.EventData.Changed
+		changesMap[host] = stats.Changed
+		changesOk := message.EventData.Ok
+		changesOk[host] = stats.Ok
+		changesFailures := message.EventData.Failures
+		changesFailures[host] = stats.Failures
+	}
+
+	return message
+}
+
+func (d *AnsibleDispatcher) RunnerOnOk(
+	runnerID string,
+	playbookFilename string,
+	playbookUUID string,
+	msgCounter int,
+	host *string,
+	tasks *results.AnsiblePlaybookJSONResultsPlayTaskItem,
+	hosts *results.AnsiblePlaybookJSONResultsPlayTaskHostsItem) *message.AnsibleRunnerJobEventYaml {
+	message := message.AnsibleRunnerJobEventYaml{
+		Event:       "v2_runner_on_ok",
+		RunnerIdent: &runnerID,
+		// StartLine:   0, //Not available
+		// EndLine:     0, //Not available
+		Created: &tasks.Duration.Start,
+		Counter: msgCounter,
+		Stdout:  hosts.Stdout,
+		EventData: &message.AnsibleRunnerJobEventYamlEventData{
+			Playbook:     &playbookFilename,
+			PlaybookUuid: &playbookUUID,
+			Host:         host,
+			TaskUuid:     &tasks.Id,
+			Task:         &tasks.Name,
+			TaskAction:   &hosts.Action,
+			Res: &message.AnsibleRunnerJobEventYamlEventDataRes{
+				Changed: &hosts.Changed,
+			},
+			Uuid: &tasks.Id,
+		},
+		Uuid: tasks.Id,
+	}
+	return &message
+}
+
+func (d *AnsibleDispatcher) RunnerOnFailed(
+	runnerID string,
+	playbookFilename string,
+	playbookUUID string,
+	msgCounter int,
+	host *string,
+	tasks *results.AnsiblePlaybookJSONResultsPlayTaskItem,
+	hosts *results.AnsiblePlaybookJSONResultsPlayTaskHostsItem) *message.AnsibleRunnerJobEventYaml {
+	message := message.AnsibleRunnerJobEventYaml{
+		Event:       "v2_runner_on_failed",
+		RunnerIdent: &runnerID,
+		// StartLine:   0, //Not available
+		// EndLine:     0, //Not available
+		Created: &tasks.Duration.Start,
+		Counter: msgCounter,
+		Stdout:  hosts.Stdout,
+		Stderr:  hosts.Stderr,
+		EventData: &message.AnsibleRunnerJobEventYamlEventData{
+			Playbook:     &playbookFilename,
+			PlaybookUuid: &playbookUUID,
+			Host:         host,
+			TaskUuid:     &tasks.Id,
+			Task:         &tasks.Name,
+			TaskAction:   &hosts.Action,
+			Res: &message.AnsibleRunnerJobEventYamlEventDataRes{
+				Changed: &hosts.Changed,
+			},
+			Uuid: &tasks.Id,
+		},
+		Uuid: tasks.Id,
+	}
+	return &message
+
+}
+
+func (d *AnsibleDispatcher) RunnerOnSkipped(
+	runnerID string,
+	playbookFilename string,
+	playbookUUID string,
+	msgCounter int,
+	host *string,
+	tasks *results.AnsiblePlaybookJSONResultsPlayTaskItem,
+	hosts *results.AnsiblePlaybookJSONResultsPlayTaskHostsItem) *message.AnsibleRunnerJobEventYaml {
+	message := message.AnsibleRunnerJobEventYaml{
```

Comment:
Maybe this could be resued for skipped/failed/on_ok, because the only difference is the Event name?